engine/test/test_path.cpp

```diff
@@ -0,0 +1,111 @@
+/* Copyright (C) 2003-2017 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "gtest/gtest.h"
+
+#include "prefix.h"
+#include "util.h"
+
+struct TestData
+{
+    const char *path;
+    const char *dir;
+    const char *base;
+};
+
+static const TestData kUnixTestData[] =
+{
+    { "/", "/", "" },
+    { "/foo", "/", "foo" },
+    { "/foo/", "/foo", "" },
+    { "/foo/bar", "/foo", "bar" },
+    { "/foo/bar/", "/foo/bar", "" },
+    { "", "", "" },
+    { "foo", "", "foo" },
+    { "foo/", "foo", "" },
+    { "foo/bar", "foo", "bar" },
+    { "foo/bar/", "foo/bar", "" },
```

Comment:
I suppose it depends exactly what the function is supposed to do. Arguably both are not actually valid paths so splitting them should fail.  At the other extreme, maybe GIGO should be applied; if you pass in something that can't possibly be a path, it doesn't matter whether the result is useful or not.  A middle ground position would be to ignore nuls and to collapse multiple slashes but only if they're at the split point, i.e. `/foo///bar/baz` -> (`/foo///bar`, `baz`) but `/foo/bar///baz` -> (`/foo/bar`, `baz`).